lib/routes/token.js

```diff
@@ -138,19 +148,14 @@ const PAYLOAD_SCHEMA = Joi.object({
 // 3. The tokens are returned in the response payload.
 module.exports = {
   validate: {
+    headers: Joi.object({
+      'authorization': Joi.string().regex(BASIC_AUTH_REGEX).optional()
+    }).options({ allowUnknown: true }),
     // stripUnknown is used to allow various oauth2 libraries to be used
     // with FxA OAuth. Sometimes, they will send other parameters that
     // we don't use, such as `response_type`, or something else. Instead
     // of giving an error here, we can just ignore them.
-    payload: function validatePayload(value, options, next) {
-      return Joi.validate(value, PAYLOAD_SCHEMA, { stripUnknown: true }, function(err, value) {
-        if (err) {
-          logger.info('routes.token.payload', err);
-        }
-
-        next(err, value);
-      });
-    }
+    payload: PAYLOAD_SCHEMA.options({ stripUnknown: true })
```

Comment:
I had to simplify this so that Hapi would invoke Joi with the correct context object to make the `$headers.authorization` reference work.  We lose logging of invalid payloads but I don't think we've been looking at those anyway.